### PR TITLE
Monkey patch get_ftp_size with dummy in SRS client

### PR DESCRIPTION
**Pull request by nabobalis**
Supersedes https://github.com/sunpy/sunpy/pull/7081

Just adds a log check for the patch.

### CODE REVIEW

File: sunpy/net/dataretriever/sources/tests/test_noaa.py [hunk 1]
```diff
@@ -257,3 +257,13 @@ def test_show():
...
+
+
+@pytest.mark.remote_data
+def test_srs_aioftp_warning(srs_client, tmpdir, caplog):
+    # For some reason it seems that the getting the size of a file is disabled on the NOAA server hence
+    # the 550 error on these calls.
+    # We now patch the fetch to remove this and we check the log is empty.
+    srs_query = srs_client.search(a.Time('2022-12-15', '2022-12-16'), a.Instrument.srs_table)
+    srs_client.fetch(srs_query, path=tmpdir)
+    assert "aioftp.errors.StatusCodeError: Waiting for ('2xx',) but got 550 [' Permission denied.']" not in caplog.text
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/tests/test_noaa.py:269]: This would fail before the patch.

**Review by Cadair — CHANGES_REQUESTED** [review]
I hate this. This is *so ugly*, surely there's a better way?

Let me have a look into parfive and a think.

**Comment by Cadair** [comment]
I have opened https://github.com/Cadair/parfive/pull/148 as an alternative to this.

**Comment by nabobalis** [comment]
So do we need to make changes in sunpy?

**Comment by Cadair** [comment]
no, just update parfive once I have done a release.

**Pull request closed without merging**